receiver.go

```diff
@@ -26,64 +26,6 @@ type Receiver struct {
 	manualCreditor *manualCreditor         // allows for credits to be managed manually (via calls to IssueCredit/DrainCredit)
 }
 
-// HandleMessage takes in a func to handle the incoming message.
-// Blocks until a message is received, ctx completes, or an error occurs.
-// When using ModeSecond, You must take an action on the message in the provided handler (Accept/Reject/Release/Modify)
-// or the unsettled message tracker will get out of sync, and reduce the flow.
-// When using ModeFirst, the message is spontaneously Accepted at reception.
-func (r *Receiver) HandleMessage(ctx context.Context, handle func(*Message) error) error {
-	debug(3, "Entering link %s Receive()", r.link.Key.name)
-
-	trackCompletion := func(msg *Message) {
-		if msg.doneSignal == nil {
-			msg.doneSignal = make(chan struct{})
-		}
-		<-msg.doneSignal
-		r.link.DeleteUnsettled(msg)
```

Comment:
this is the reason for the handler.
you can only remove the message from the map tracking it on settlement, and not before.
otherwise you are receiving an unlimited messages concurrently, no matter the link credit tha tyou initially configure, because the map tracking the message being handled is cleared as soon as you return the message.

Reply:
This is an interesting one. There are two credit management algorithms now (manual and the previous one). 

I think we _only_ need to activate this logic when we're **not** doing manual credit management. So I should have extended some additional changes (probably to just not track these 'unsettled' messages in the first place if a manualCreditor exists).

Reply:
Added a check to not auto-accept the message when manual credit management is active.